### PR TITLE
[clear] Replace shell=True, replace exit() by sys.exit()

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

#### What I did
`subprocess()` - when using with `shell=True` is dangerous. Using subprocess function without a static string can lead to command injection.
`sys.exit` is better than `exit`, considered good to use in production code.
Ref:
https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python
https://stackoverflow.com/questions/19747371/python-exit-commands-why-so-many-and-when-should-each-be-used
#### How I did it
`subprocess()` - use `shell=False` instead, use list of strings Ref: [https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation](https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation)
Replace `exit()` by `sys.exit()`
#### How to verify it
Manual test, execute all `sonic-clear` commands
Add UT

### CODE REVIEW

File: clear/main.py [hunk 1]
```diff
@@ -79,16 +79,14 @@ def get_command(self, ctx, cmd_name):
 # location (configdb?), so that we prevent the continous execution of this
 # bash oneliner. To be revisited once routing-stack info is tracked somewhere.
 def get_routing_stack():
-    command = "sudo docker ps | grep bgp | awk '{print$2}' | cut -d'-' -f3 | cut -d':' -f1"
+    cmd0 = ["sudo", "docker", "ps"]
+    cmd1 = ["grep", "bgp"]
+    cmd2 = ["awk", '{print$2}']
+    cmd3 = ["cut", "-d", '-', "-f3"]
```
Comment (qiluo-msft) [inline clear/main.py:85]: >"-d", '-' [](http://example.com/codeflow?start=19&length=9)

Originally it is one arg.

File: clear/main.py [hunk 2]
```diff
@@ -79,16 +79,14 @@ def get_command(self, ctx, cmd_name):
...
+    cmd4 = ["cut", "-d", ':', "-f1"]
```
Comment (qiluo-msft) [inline clear/main.py:86]: >"-d", ':' [](http://example.com/codeflow?start=19&length=9)

Originally it is one arg.

File: clear/main.py (hunk 1 above)
Comment (maipbui) [reply clear/main.py]: the result is the same as original behavior, but I did change to one arg.

Comment (qiluo-msft) [reply clear/main.py]: Just wandering, is there any diff between `-d-` and `-d'-'` ?

Comment (maipbui) [reply clear/main.py]: if use direct command in linux, then no difference. But in this case `cmd3 = ["cut", "-d'-'", "-f3"]` will lead to an error, cut: the delimiter must be a single character. So we need to use either `cmd3 = ["cut", "-d-", "-f3"]` or `cmd3 = ["cut", "-d", '-', "-f3"]`

Comment (qiluo-msft) [reply clear/main.py]: Make sense. Bash handles the quotation marks. `shell=False` bypassed the shell.

**Review by qiluo-msft — APPROVED** [review]

File: clear/bgp_frr_v6.py [hunk 3]
```diff
@@ -69,22 +69,22 @@ def soft_in(ipaddress):
...
+        command = ['sudo', 'vtysh', '-c', "clear bgp ipv6 {} soft in".format(ipaddress)]
     else:
-        command = 'sudo vtysh -c "clear bgp ipv6 * soft in"'
+        command = ['sudo', 'vtysh', '-c', "clear bgp ipv6 * soft in"]
     run_command(command)
 
 
 # 'soft all' subcommand
-@neighbor.command('all')
+@soft.command('all')
```
Comment (qiluo-msft) [inline clear/bgp_frr_v6.py:79]: >soft [](http://example.com/codeflow?start=1&length=4)

Thanks for the bug fix!

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**